### ISSUE TITLE
[BUG] ICE: Load/Store operand must be a pointer when loading/storing aligned FP4 tensor

### ISSUE THREAD

**Issue by Evan704** [NVIDIA/cutlass#3349]
### Which component has the problem?

CuTe DSL

### Bug Report

**Describe the bug**
An ICE occurs when attempting to `.load()` and `.store()` an aligned `Float4E2M1FN` tensor using CuTe DSL. Even though the total bits are 32-bit aligned, the compiler generates an invalid LLVM IR struct (`{ ptr, i8 }`) for the load/store operands, causing the LLVM Verifier to fail with `Load/Store operand must be a pointer`.

**Code to reproduce bug**
```python
import cutlass
import cutlass.cute as cute
import torch
from cutlass.cute.runtime import from_dlpack

@cute.jit
def test(a: cute.Tensor):
    a_vec = a.load()
    a.store(a_vec)

a = torch.zeros(4, 4, dtype=torch.float4_e2m1fn_x2)
a_ = from_dlpack(a)
test(a_)
```

**Expected behavior**
The compiler should emit correct bit-shifts/masks or lower the i4 sub-byte elements through an i32 register without breaking the LLVM verifier.

**Environment details**
 - **GPU**: RTX 5090
 - **CUTLASS**: 4.5.2
 - **CUDA**: 13.1.2

**Additional context**
If the tensor is 1D (e.g., `torch.zeros(4, dtype=...)`), the code compiles successfully.

**Comment by brandon-yujie-sun** [comment]
Thanks for reporting this, looks like a bug for sub-byte handling. Will take a look.

**Comment by brandon-yujie-sun** [comment]
@Evan704 This is fixed in 4.6.2 and 4.7
